syntax = "proto3";

package RecipeRecommendationGRPC;

message Recipe {
  string id = 1;
  string title = 2;
  string ingredients = 3;
  string steps = 4;
}

service RecipeRecommendation {
  rpc GetAllRecipes(GetAllRecipesRequest) returns (GetAllRecipesResponse);
  rpc UpdateRecipe(UpdateRecipeRequest) returns (UpdateRecipeResponse);
  rpc DeleteRecipe(DeleteRecipeRequest) returns (DeleteRecipeResponse);
  rpc CreateRecipe(CreateRecipeRequest) returns (CreateRecipeResponse);
}

message GetAllRecipesRequest {
  
}

message GetAllRecipesResponse {
  repeated Recipe recipes = 1;
}

message CreateRecipeRequest {
  Recipe recipe = 1;
}

message CreateRecipeResponse {
  Recipe recipe = 1;
}

message UpdateRecipeRequest {
  string id = 1;
  Recipe recipe = 2;
}

message UpdateRecipeResponse {
  Recipe recipe = 1;
}

message DeleteRecipeRequest {
  string id = 1;
}

message DeleteRecipeResponse {
  string message = 1;
}



